### ISSUE TITLE
Authentication exceptions don't include details

### ISSUE THREAD

**Issue by GCorbel** [sferik/x-ruby#3]
When I try to use an endpoint with no access_token_secret, I receive an authentication exception as expected but `X::Errors::Error#object` is `nil`. It happens because the content type returned is "application/problem+json" and not "application/json; charset=utf-8" as defined in `X::ClientDefaults#DEFAULT_CONTENT_TYPE`.

I suppose `X::Errors::Error#object` should return the payload given.

**Comment by sferik** [comment]
Thanks for reporting this bug @GCorbel! I will try to fix it as quickly as possible. Can you please provide details about the request you’re making that is causing this error response? If it’s one of the non-free endpoints, I might need your help to verify the fix.

For what it’s worth, I just applied for [GitHub Sponsors](https://github.com/sponsors/sferik/dashboard). If you (or your company) would be willing to sponsor me to the tune of $100/month, I’d put 100% of that toward paying for my own Twitter API Basic account, so I could do my own testing and ensure this feature works.

**Comment by sferik** [comment]
What version of the library are you using? In the latest version, initializing an `X::Client` without an `access_token_secret` produces the following error upon initialization and won’t allow you to attempt a request without one:

```
…x-ruby/lib/x/client.rb:23:in `initialize': missing keyword: :access_token_secret (ArgumentError)
```

**Comment by GCorbel** [comment]
I was able to reproduce the issue with 0.7.1. I didn't mention that access_token_secret have to be an empty string and not `nil`.

```
client = X::Client.new(
  api_key: 'something',
  api_key_secret: 'something',
  access_token: 'something',
  access_token_secret: '',
)
begin
  client.get('users/me')
  # `response['content-type']` in X::Error#initialize is "application/problem+json"
rescue => e
  e.object # => nil
end
```

**Comment by sferik** [comment]
This should be fixed in gem version 0.8.0, which I just released.

Thank you for providing sample code. When I run your code in the latest version, I get the following output:

```ruby
{"title"=>"Unauthorized", "type"=>"about:blank", "status"=>401, "detail"=>"Unauthorized"}
```
